docs/README.md

```diff
@@ -19,417 +17,498 @@
+- [Notify author of failed guidelines](#issues) when opening an issue.
+- Schedule [detection stale issues and pull requests](#staleness) and notify author and collaborators.
+- And [more](#configuration)
 
-## Key Features
+---
 
-- [Prevent pull requests to be merged](#pull-requests)  based on configured rulesets
-- [Notify author of failed guidelines when openning an issue](#issues) based on configured rulesets
-- [Detect stale issues and pull requests](#staleness) and notify author and collaborators.
+**📖 Contents:** [Usage](#usage) ◦ [Configuration](#configuration) ◦ [Roadmap](#roadmap) ◦ [Support](#support) ◦ [Contributions](#contributions) ◦ [Authors](#authors)
 
-### Pull Requests
+---
 
-Validate pull requests for mergeability based on content and structure of your PR (title, labels, milestone, project, description, approvals, etc). The rule-sets are fully configurable. Here are a few examples of what you can do:
 
-- Prevent accidental merging of Pull Requests that are work in progress by labeling it `wip` or prefixing the title with the abbreviation.
+# Usage
 
-- Ensure all Pull Requests have a description so that when you view through history you still have context.
+1. [Install](https://github.com/apps/mergeable) the Mergeable GitHub App.
+2. [Create](#configuration) your recipe(s). Here are some [examples](#examples).
+3. Commit and push the recipes to your repository at `.github/mergeable.yml`
 
-- Ensure that all Pull Requests are approved by a specific list of users. It is especially useful if one of the users is not a collaborator of your repository -- something GitHub does not already support.
+> ☝ **NOTE:** You can also [deploy to your own server](deploy.md).
 
-- Ensure that all Pull Requests merged are in a specific GitHub Project. Mergeable even detects when you are closing an issue that is associated with the right project. This is very useful when your process includes QA validation of stories.
+# Configuration
 
-- [And more](#configuration).
+**Mergeable** is **highly** configurable.
+Define your recipes by creating a `.github/mergeable.yml` file in your repository.
 
-### Issues
+## Basics
+The configuration consists of any number of recipes. Recipes are created by tying [events](#events) with a set of [validators](#validators) and [actions](#actions) together:
 
-There may be certain formats and structure you want your Github issues to adhere to. Mergeable allows you to configure this and will create a comment with a list of suggested improvements to your Issue.
+```yml
+version: 2
+mergeable:
+  - when: {{event}}, {{event}} # can be one or more
+    validate:
+      # list of validators. Specify one or more.
+      - do: {{validator}}
+        {{option}}: # name of an option supported by the validator.
+          regex: {{expression}} # where regex is a nested option
+    pass: # list of actions. Specify one or more. Omit this tag if no actions are needed.
```

Comment:
i think it might be good to explicitly say `list of actions to be taken if all the validators returned true`
and there is also a hierarchy here which may not be obviously stated. Which are (in highest to lowest order)
If some validators returned error : do `error` action 
If some validators failed and no errors: do `fail` action
If all validator passed : do `pass` action

Reply:
Updated as you suggested. LMK what you think.